### PR TITLE
`Development`: Add more communication e2e tests

### PR DESCRIPTION
**Pull request by TheZoker**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).

### Description

This PR adds more communication tests:

Course communicaton:
- Instructor pins a post
- Instructor archives a post
- Instructor selects answer
- Student reacts to post

Exercise communication:
- Student creates post within exercise
- Student searches for exercise
- Student replies to post within exercise
- Student reacts to post within exercise

#### Code Review
- [x] Review 1
- [ ] Review 2

### CODE REVIEW

File: src/test/cypress/e2e/course/CourseCommunication.cy.ts [hunk 1]
```diff
@@ -39,24 +40,70 @@ describe('Course communication', () => {
...
+            courseCommunication.setTitle(title);
+            courseCommunication.setContent(content);
+            courseCommunication.save();
+            cy.login(studentTwo, `/courses/${course.id}/discussion`);
+            navigationBar.openNotificationPanel();
+            navigationBar.getNotifications().first().find('.notification-title').contains('New announcement');
+            navigationBar.getNotifications().first().find('.notification-text').contains(`Course "${course.title}" got a new announcement.`);
+        });
+
+        it('instructor should be able archive a post', () => {
```
Comment (frederik-kt) [inline src/test/cypress/e2e/course/CourseCommunication.cy.ts:78]:
```suggestion
        it('instructor should be able to archive a post', () => {
```

**Review by frederik-kt — APPROVED** [review]
Code LGTM

**Review by tobias-lippert — APPROVED** [review]
Code LGTM

**Review by DominikRemo — DISMISSED** [review]
LGTM

**Review by DominikRemo — APPROVED** [review]
reapprove

**Review by frederik-kt — APPROVED** [review]
Reapprove code

File: src/test/cypress/e2e/course/CourseCommunication.cy.ts [hunk 2]
```diff
@@ -143,6 +203,64 @@ describe('Course communication', () => {
...
+
+        it('other students should be able to react to exercise post', () => {
+            const title = 'My React Test Post';
+            const content = 'Test React Post Content';
+            cy.login(studentOne, `/courses/${course.id}/discussion`);
+            courseManagementRequest.createCourseExercisePost(course, textExercise, title, content).then((response) => {
+                const post = response.body;
+                cy.login(studentTwo, `/courses/${course.id}/exercises/${textExercise.id}`);
+                cy.reload();
+                courseCommunication.react(post.id, 'thumbs up');
```
Comment (bassner) [inline src/test/cypress/e2e/course/CourseCommunication.cy.ts:259]: For ``reply`` and ``react``, as far as I see you only check if there's a button they can press to actually perform the action. Then you click it. However, shouldn't there be some sort of verification if the action has actually worked (e.g. reload, then check for the reaction)?

File: src/test/cypress/e2e/course/CourseCommunication.cy.ts [hunk 3]
```diff
@@ -115,6 +162,19 @@ describe('Course communication', () => {
...
+        it('other students should be able to react to post', () => {
+            const title = 'My React Test Post';
+            const content = 'Test React Post Content';
+            const context = CourseWideContext.RANDOM;
+            cy.login(studentOne, `/courses/${course.id}/discussion`);
+            courseManagementRequest.createCoursePost(course, title, content, context).then((response) => {
+                const post = response.body;
+                cy.login(studentTwo, `/courses/${course.id}/discussion`);
+                cy.reload();
+                courseCommunication.react(post.id, 'thumbs up');
```
Comment (bassner) [inline src/test/cypress/e2e/course/CourseCommunication.cy.ts:174]: Same here

File: src/test/cypress/e2e/course/CourseCommunication.cy.ts [hunk 4]
```diff
@@ -39,24 +40,70 @@ describe('Course communication', () => {
...
                 cy.login(studentTwo, `/courses/${course.id}/discussion`);
-                navigationBar.openNotificationPanel();
-                navigationBar.getNotifications().first().find('.notification-title').contains('New announcement');
-                navigationBar.getNotifications().first().find('.notification-text').contains(`Course "${course.title}" got a new announcement.`);
+                courseManagementRequest.createCoursePostReply(course, post, 'Answer Reply').then((response) => {
+                    const answerPost = response.body;
+                    cy.login(instructor, `/courses/${course.id}/discussion`);
+                    cy.reload();
+                    courseCommunication.showReplies(post.id);
+                    courseCommunication.markAsAnswer(answerPost.id);
```
Comment (bassner) [inline src/test/cypress/e2e/course/CourseCommunication.cy.ts:105]: Same here

**Review by bassner — CHANGES_REQUESTED** [review]
Just one thing I've been asking myself :)

File: src/test/cypress/e2e/course/CourseCommunication.cy.ts (hunk 2 above)
Comment (TheZoker) [reply src/test/cypress/e2e/course/CourseCommunication.cy.ts]: Good point! I initially wanted to add that within a follow-up PR, but it makes more sense to include it in this one. I added the requested checks in the latest commit. Thanks for the hint!

**Review by bassner — APPROVED** [review]
Thanks for the updates. LGTM 👍

**Pull request merged**